tests/foreman/ui/test_hardwaremodel.py

```diff
@@ -55,9 +59,24 @@ def test_positive_end_to_end(session, module_org, module_location):
         assert hm_values['vendor_class'] == vendor_class
         assert hm_values['info'] == info
         # Create host with associated hardware model
-        host_name = create_fake_host(
-            session, host_template, extra_values={'additional_information.hardware_model': name}
+        host = make_fake_host(
```

Comment:
This does feel like a test step for the `end to end` context of this test, and as such something that should be done via the UI.

Reply:
I could be convinced otherwise, since at least the update to the host to set the hardware model is done through the UI. But even then, this is changing the test behavior to update the host with hardware model - instead of specifying it on host creation.  

I'd take one more step back with this test function, and look at what its doing in general.

The `host_template` creation should get moved into a fixture, its not part of the test coverage and is done via API anyway.

Reply:
I think the question is whether or not the fake host creation is considered part of the coverage. Something the component owner needs to decide

Reply:
@mshriver @JacobCallahan Our objective to update the hardware model with host is achieved here. Do we need the host to be created through UI only?